src/Codata/Delay/Properties.agda

```diff
@@ -48,3 +49,64 @@ module _ {a b c} {A : Set a} {B : Set b} {C : Set c} where
  map-unfold-fusion f n s with n s
  ... | Sum.inj₁ s′ = later λ where .force → map-unfold-fusion f n s′
  ... | Sum.inj₂ b  = now Eq.refl
+
+
+⇓-unique : ∀ {a} → {A : Set a} →
+           {d : Delay A ∞} →
+           (d⇓₁ : d ⇓) → (d⇓₂ : d ⇓) →
+           d⇓₁ ≡ d⇓₂
+⇓-unique {d = now s} (now s) (now s) = Eq.refl
+⇓-unique {d = later d'} (later l) (later r) =
+  Eq.cong later (⇓-unique {d = force d'} l r)
+
+
+bind⇓-injₗ : ∀ {a} {A B : Set a} →
+             {d : Delay A ∞} {f : A → Delay B ∞} →
+             bind d f ⇓ → d ⇓
+bind⇓-injₗ {d = now s} foo = now s
+bind⇓-injₗ {d = later s} (later foo) =
+  later (bind⇓-injₗ foo)
+
+
+bind⇓-injᵣ : ∀ {a} {A B : Set a}
+             {d : Delay A ∞} {f : A → Delay B ∞} →
+             (bind⇓ : bind d f ⇓) →
+             f (extract (bind⇓-injₗ {d = d} {f = f} bind⇓)) ⇓
+bind⇓-injᵣ {d = now s} foo = foo
+bind⇓-injᵣ {d = later s} {f} (later foo) =
+  bind⇓-injᵣ {d = force s} {f = f} foo
+
+
+-- The extracted value of a bind is equivalent to the extracted value of its
+-- second element
+extract-bind-⇓ : ∀ {a} → {A B : Set a}
+               {x : Delay A Size.∞} → {f : A → Delay B Size.∞} →
+               (x⇓ : x ⇓) → (f⇓ : ((f (extract x⇓)) ⇓)) →
```

Comment:
Can this be just `(f⇓ : (f (extract x⇓)) ⇓)`?

Reply:
Yup :)